src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java

```diff
@@ -302,28 +304,150 @@ public void setOption(String optionStr) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST parent = ast.getParent();
-        // we do not want to check colon for cases and defaults
-        if (parent.getType() != TokenTypes.LITERAL_DEFAULT
-                && parent.getType() != TokenTypes.LITERAL_CASE) {
-            final String text = ast.getText();
-            final int colNo = ast.getColumnNo();
-            final int lineNo = ast.getLineNo();
-            final String currentLine = getLine(lineNo - 1);
-
-            // Check if rest of line is whitespace, and not just the operator
-            // by itself. This last bit is to handle the operator on a line by
-            // itself.
+        if (isOperatorNode(ast)) {
             if (option == WrapOption.NL
-                    && !text.equals(currentLine.trim())
-                    && CommonUtil.isBlank(currentLine.substring(colNo + text.length()))) {
-                log(ast, MSG_LINE_NEW, text);
+                    && TokenUtil.areOnSameLine(ast, getLeftNode(ast))
+                    && !TokenUtil.areOnSameLine(ast, getRightNode(ast))) {
+                log(ast, MSG_LINE_NEW, ast.getText());
             }
             else if (option == WrapOption.EOL
-                    && CommonUtil.hasWhitespaceBefore(colNo - 1, currentLine)) {
-                log(ast, MSG_LINE_PREVIOUS, text);
+                    && !TokenUtil.areOnSameLine(ast, getLeftNode(ast))) {
+                log(ast, MSG_LINE_PREVIOUS, ast.getText());
+            }
+        }
+    }
+
+    /**
+     * Filters some false tokens that this check should ignore.
+     *
+     * @param node the node to check
+     * @return {@code true} for all nodes this check should validate
+     */
+    private static boolean isOperatorNode(DetailAST node) {
```

Comment:
description of method is not same a name of method.
should we name method more abstract ... like `isTargetNode`

Reply:
done